common/utils/useWhatHasUpdated.ts

```diff
@@ -0,0 +1,222 @@
+import React from 'react';
+
+let what_debug_changed = 0;
+
+/**
+ * stackoverflow random color logic
+ */
+const getRandomColor = () => {
+  var letters = '0123456789ABCDEF';
+  var color = '#';
+  for (var i = 0; i < 6; i++) {
+    color += letters[Math.floor(Math.random() * 16)];
+  }
+  return color;
+};
+
+/**
+ *
+ * Check whether the dependency item is an object.
+ */
+const isObject = (t: any) =>
+  Object.prototype.toString.call(t) === '[object Object]';
+
+const getPrintableInfo = (dependencyItem: any) => {
+  /**
+   * Printing the info into viewable format
+   */
+  if (isObject(dependencyItem) || Array.isArray(dependencyItem)) {
+    let ans;
+    try {
+      ans = JSON.stringify(dependencyItem, null, 2);
+    } catch (e) {
+      ans = 'CIRCULAR JSON';
+    }
+    return ans;
+  }
+
+  return dependencyItem;
+};
+
+const useHotRefs = (value: any) => {
+  const fnRef = React.useRef(value);
+  React.useEffect(() => {
+    fnRef.current = value;
+  });
+
+  return fnRef;
+};
```

Comment:
Can we try to avoid using `any`s? I think it could be something like

```typescript
const useHotRefs = <T>(value: T): MutableRefObject<T> => {
 const fnRef = React.useRef<T>(value);
  React.useEffect(() => {
    fnRef.current = value  
  }):
	return fnRef;
}
```